modules/services/window-managers/i3-sway/lib/functions.nix

```
@@ -77,88 +77,70 @@ rec {
     let colorsNotNull = lib.filterAttrs (n: v: v != null) colors != { };
     in ''
       bar {
-        ${optionalString (id != null) "id ${id}"}
-        ${fontConfigStr fonts}
-        ${optionalString (mode != null) "mode ${mode}"}
-        ${optionalString (hiddenState != null) "hidden_state ${hiddenState}"}
-        ${optionalString (position != null) "position ${position}"}
         ${
-          optionalString (statusCommand != null)
-          "status_command ${statusCommand}"
+          concatStringsSep "\n" (notEmpty [
+            "${optionalString (id != null) "id ${id}"}"
```

There is no need to wrap any of these `optionalString`s in another string:

```suggestion
            (optionalString (id != null) "id ${id}")
```

---

Thanks for the feedback, will adjust and fix the tests just wanted to get sure that I'm not on the wrong path...
Would it be useful to have something like not Empty in lib or is there something similar?

---

I think it makes sense to move `notEmpty` to lib so that we can take advantage of it across the codebase. I don't know of anything like that in nixpkgs lib or hm lib.

---

@sumnerevans ping

---

@sumnerevans bump?